distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/MachineDependentHeap.java

```diff
@@ -8,193 +8,153 @@
             }
-        }
-
-        @SuppressWarnings("unchecked")
-        private static <T> boolean containsOnly(Collection<T> collection, T... items) {
-            return Arrays.asList(items).containsAll(collection);
-        }
+        };
     }
 
-    enum MachineNodeRole {
-        /**
-         * Master-only node.
-         *
-         * <p>Heap is computed as 60% of total system memory up to a maximum of 31 gigabytes.
-         */
-        MASTER_ONLY(m -> mb(min((long) (m * .6), MAX_HEAP_SIZE))),
-
-        /**
-         * Machine learning only node.
-         *
-         * <p>Heap is computed as:
-         * <ul>
-         *     <li>40% of total system memory when total system memory 16 gigabytes or less.</li>
-         *     <li>40% of the first 16 gigabytes plus 10% of memory above that when total system memory is more than 16 gigabytes.</li>
-         *     <li>The absolute maximum heap size is 31 gigabytes.</li>
-         * </ul>
-         *
-         * In all cases the result is rounded down to the next whole multiple of 4 megabytes.
-         * The reason for doing this is that Java will round requested heap sizes to a multiple
-         * of 4 megabytes (certainly versions 11 to 18 do this), so by doing this ourselves we
-         * are more likely to actually get the amount we request. This is worthwhile for ML where
-         * the ML autoscaling code needs to be able to calculate the JVM size for different sizes
-         * of ML node, and if Java is also rounding then this causes a discrepancy. It's possible
-         * that a future version of Java could round to an even bigger number of megabytes, which
-         * would cause a discrepancy for people using that version of Java. But there's no harm
-         * in a bit of extra rounding here - it can only reduce discrepancies.
-         *
-         * If this formula is changed then corresponding changes must be made to the {@code NativeMemoryCalculator} and
-         * {@code MlAutoscalingDeciderServiceTests} classes in the ML plugin code. Failure to keep the logic synchronized
-         * could result in repeated autoscaling up and down.
-         */
-        ML_ONLY(m -> mb(m <= (GB * 16) ? (long) (m * .4) : (long) min((GB * 16) * .4 + (m - GB * 16) * .1, MAX_HEAP_SIZE), 4)),
-
-        /**
-         * Data node. Essentially any node that isn't a master or ML only node.
-         *
-         * <p>Heap is computed as:
-         * <ul>
-         *     <li>40% of total system memory when less than 1 gigabyte with a minimum of 128 megabytes.</li>
-         *     <li>50% of total system memory when greater than 1 gigabyte up to a maximum of 31 gigabytes.</li>
-         * </ul>
-         */
-        DATA(m -> mb(m < GB ? max((long) (m * .4), MIN_HEAP_SIZE) : min((long) (m * .5), MAX_HEAP_SIZE))),
-
-        /**
-         * Unknown role node.
-         *
-         * <p>Hard-code heap to a default of 1 gigabyte.
-         */
-        UNKNOWN(m -> DEFAULT_HEAP_SIZE_MB);
```

Comment:
Removing this default is deliberate?

Reply:
Yes. It was only possible because we were parsing elasticsearch.yml manually here before. Now we use the parsed role, and anything but ML or master only is data.